lib/new.js

```diff
@@ -160,19 +161,54 @@ const promptForName = (args) => {
     return value;
   };
 
-  // No need to check for err since we're using a validator
-  const handler = (err, value) => {
-    settings.spa = value;
-    settings.name = 'skyux-spa-' + value;
-    settings.path = path.join('.', settings.name);
-    settings.pathTmp = path.join(settings.path, 'tmp');
-    promptForUrl();
-  };
+  return promptly.prompt(prompt, { validator: validator })
+    .then((value) => {
+      settings.spa = value;
+      settings.name = 'skyux-spa-' + value;
+      settings.path = path.join('.', settings.name);
+      settings.pathTmp = path.join(settings.path, 'tmp');
+      return Promise.resolve(value);
+    });
+};
+
+/**
+ * Returns an object representing the template config, derived
+ * from constructor arguments.
+ * @name getTemplateFromArgs
+ */
+const getTemplateFromArgs = (args) => {
```

Comment:
@Blackbaud-SteveBrush I'm confident your code works here, but I wonder if we could simplify it for readability.

```
  let name = 'default';
  let url = 'https://github.com/blackbaud/skyux-template';

  if (args.t) {
    args.template = args.t;
  }

  if (args.template) {
    name = args.template;
    url = `${url}-${name}`;
  }

  return Object.freeze({
    name: name,
    url: url
  });
```

The `args` passed to this method have already been [processed by minimist](https://github.com/blackbaud/skyux-cli/blob/master/bin/skyux.js#L12), which means `{}` will always exist.